luigi/parameter.py

```diff
@@ -137,11 +137,7 @@ def __init__(self, default=_no_value, is_boolean=False, is_global=False, signifi
                                 as positional arguments become very tricky when
                                 you have inheritance and whatnot.
         """
-        if is_boolean:
-            self.__default = False
```

Comment:
You still need this code path in some way. If somebody doesn't explicitly set `default=True/False`, it should default to false.

I think it should be something like `self.__default = default or False`. As long as the tests pass (without you changing the current tests :)), it should be OK.